open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/model/PeriodicLimit.java

```diff
@@ -0,0 +1,173 @@
+package com.wso2.openbanking.accelerator.consent.extensions.manage.model;
+
+import com.wso2.openbanking.accelerator.consent.extensions.common.ConsentExtensionConstants;
+import com.wso2.openbanking.accelerator.consent.extensions.util.PeriodicTypesEnum;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.*;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAdjusters;
+
+/**
+ * This class represents a periodic limit for a VRP consent.
+ * It includes the period type, amount, period alignment, cyclic expiry time, and cyclic paid amount.
+ */
+public class PeriodicLimit {
+    private final String periodType;
+    private BigDecimal amount;
+    private String periodAlignment;
+    private long cyclicExpiryTime;
+    private BigDecimal cyclicRemainingAmount;
+
+    /**
+     * Constructs a new PeriodicLimit with the specified period type, amount, and period alignment.
+     * It also calculates and sets the cyclic expiry time and cyclic paid amount.
+     *
+     * @param periodType the period type
+     * @param amount the amount
+     * @param periodAlignment the period alignment
+     */
+    public PeriodicLimit(String periodType, BigDecimal amount, String periodAlignment) {
+        this.periodType = periodType;
+        this.amount = amount;
+        this.periodAlignment = periodAlignment;
+        setCyclicExpiryTime();
+        calculateCyclicPaidAmount();
+    }
+
+    public BigDecimal getAmount() {
+        return amount;
+    }
+
+    public void setAmount(BigDecimal amount) {
+        this.amount = amount;
+    }
+
+    public String getPeriodAlignment() {
+        return periodAlignment;
+    }
+
+    public void setPeriodAlignment(String periodAlignment) {
+        this.periodAlignment = periodAlignment;
+    }
+
+    public long getCyclicExpiryTime() {
+        return cyclicExpiryTime;
+    }
+
+    public BigDecimal getCyclicRemainingAmount() {
+        return cyclicRemainingAmount;
+    }
+
+    public void setCyclicRemainingAmount(BigDecimal cyclicRemainingAmount) {
+        this.cyclicRemainingAmount = cyclicRemainingAmount;
+    }
+
+    /**
+     * Calculates and sets the cyclic expiry time based on the period type and period alignment.
+     */
+    public void setCyclicExpiryTime() {
+        Instant now = Instant.now();
+        Instant expiryTime;
+
+        if (periodAlignment.equalsIgnoreCase(ConsentExtensionConstants.CONSENT)) {
```

Comment:
Do we need to support either case here, shall we stick to the spec name and use equals()